### PR TITLE
fix comment for parameters "end_stream" of decodeData/encodeData.

### PR DESCRIPTION
**Pull request by wangfakang**
Commit Message: fix comment for parameters end_stream of decodeData/encodeData.
Risk Level: Low
Docs Changes: Updates the file include/envoy/http/filter.h
@mattklein123 plz help review,thx.
The relevant code is implemented as [follows](https://github.com/envoyproxy/envoy/blob/master/source/common/http/filter_manager.cc#L604), which is inconsistent with the function comments in [filter.h](https://github.com/envoyproxy/envoy/blob/master/include/envoy/http/filter.h#L593) file.

```
void FilterManager::decodeData(ActiveStreamDecoderFilter* filter, Buffer::Instance& data,
                               bool end_stream,
                               FilterIterationStartState filter_iteration_start_state) {
……
    (*entry)->end_stream_ = end_stream && !filter_manager_callbacks_.requestTrailers();
    FilterDataStatus status = (*entry)->handle_->decodeData(data, (*entry)->end_stream_);
……
}
```

### CODE REVIEW

File: include/envoy/http/filter.h [hunk 1]
```diff
@@ -591,6 +591,7 @@ class StreamDecoderFilter : public StreamFilterBase {
    * Called with a decoded data frame.
    * @param data supplies the decoded data.
    * @param end_stream supplies whether this is the last data frame.
+   * Further note that end_stream is only true if there are no trailers.
```
Comment (htuch) [inline include/envoy/http/filter.h:594]: Isn't this a tautology? I.e. the fact that the data frame is last already means this.

Comment (wangfakang) [reply include/envoy/http/filter.h:594]: > Isn't this a tautology? I.e. the fact that the data frame is last already means this.

@htuch The data frame is easily mistaken for the body of the request or response. Here's the same [issue](https://github.com/envoyproxy/envoy/issues/12196).
```
virtual FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) PURE
```

Comment (htuch) [reply include/envoy/http/filter.h:594]: Fair, I guess this is clearer.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**